fault/actions.py

```diff
@@ -249,6 +250,40 @@ def retarget(self, new_circuit, clock):
                     actions=actions, count=self.count)
 
 
+class Fork(Action):
+    def __init__(self, name, actions):
+        self.name = name
+        self.actions = actions
+
+    def __str__(self):
+        return f"Fork({self.name}"
+
+    def retarget(self, new_circuit, clock):
+        actions = [action.retarget(new_circuit, clock) for action in
+                   self.actions]
+        return Fork(self.name, actions)
+
+
+class JoinType(enum.Enum):
+    Default = enum.auto()
+    None_ = enum.auto()
+    Any = enum.auto()
+
+
+class Join(Action):
+    def __init__(self, *processes, join_type=JoinType.Default):
+        self.join_type = join_type
+        self.processes = list(*processes)
```

Comment:
Does this need to be converted to a list? By default `*args` will be a tuple, so we can avoid the conversion unless we plan to use the mutability